src/actions/blockchain.ts

```diff
@@ -161,8 +164,110 @@ const errorHandling = (error: Error) => async (dispatch: Function, getState: Fun
   }))
 }
 
+/**
+ * checkEthTokenBalance > returns false or EtherToken Balance
+ * @param token 
+ * @param weiSellAmount 
+ * @param account 
+ * @returns boolean | BigNumber <false, amt>
+ */
+const checkEthTokenBalance = async (token: TokenCode, weiSellAmount: BigNumber, account?: Account): Promise<boolean | BigNumber> => { 
+  // perform checks
+  // return if token is not ETHER
+  if (token !== "ETH") return false
+  const wrappedETH = await getEtherTokenBalance(token, account)
+  // return false if wrapped Eth is enough
+  if (wrappedETH.gte(weiSellAmount)) return false
+
+  return weiSellAmount.minus(wrappedETH)
+}
+
+/**
+ * checkTokenAllowance > returns false or Token[name] Allowance
+ * @param token 
+ * @param weiSellAmount 
+ * @param account 
+ * @returns boolean | BigNumber <false, amt>
+ */
+const checkTokenAllowance = async (token: TokenCode, weiSellAmount: BigNumber, account?: Account): Promise<boolean | BigNumber> => { 
+  // perform checks
+  const tokenAllowance = await getTokenAllowance(token, account)
+  // return false if wrapped Eth is enough
+  if (tokenAllowance.gte(weiSellAmount)) return false
+
+  return tokenAllowance
+}
+
+/** 
+ * checkUserStateAndSell()(dispatch, state) => THUNK Action
+ * 
+*/
+export const checkUserStateAndSell = () => async (dispatch: Function, getState: Function) => {
+  const { tokenPair: { sell, sellAmount }, blockchain: { activeProvider, currentAccount } } = getState()
+  const weiSellAmt = await toWei(sellAmount)
+  
+  try {
+    // change to modal with button, new modal
+    dispatch(openModal({
+      modalName: 'TransactionModal',   
+      modalProps: {
+        header: `Contacting Ethereum blockchain`,
+        body: `Please wait`,
+      },
+    }))
+    // check ETHER deposit && allowance amount
+    const wrappedETH = await checkEthTokenBalance(sell, weiSellAmt, currentAccount)
+    const tokenAllowance = await checkTokenAllowance(sell, weiSellAmt, currentAccount)
```

Comment:
Start allowance request here, but `await` on L234, no need to block execution here.